### DISCUSSION TITLE
en/stable/examples/workflows/upstream/workflow_template__steps/

### DISCUSSION THREAD

**Discussion by giscus** [argoproj-labs/hera#1091] [Website Discussions]
# en/stable/examples/workflows/upstream/workflow_template__steps/

https://hera.readthedocs.io/en/stable/examples/workflows/upstream/workflow_template__steps/

**Comment by jiantang99** [comment]
If I wanted to have multiple parameters passed in per step, do I used a list? I.e., arguments=[Parameter(name="message1", value="hello1b"),Parameter(name="message2", value="hello2b"),]
Or should I be using another way?

**Reply by elliotgunton** [reply]
You can use a list as you've written, or you can pass a dictionary of key-values: `arguments={"message1": "hello1b", "message2": "hello2b"}`.

**Comment by gentry-au** [comment]
I am starting to use this framework to build out some CI/CD pipelines. I'm really enjoying Hera. We are triggering a workflow based on PR events and wanted to define something like:

github webhook -> sensor (trigger workflow via workflowtemplate)-> workflow runs

The workflow is created perfectly but now I was thinking to make the webhook/sensor IaC since we manually created these. It seems that the tool is geared towards workflows only. I do see all of the classes defined (e.g. Sensor, SensorSpec, etc.), but I don't see how to connect all of these. Is there a way to do this using Hera?

```
    sensor = Sensor(
        spec=SensorSpec(
            dependencies=[
                EventDependency(
                    name="github-events",
                    event_source_name="github-events",
                    event_name="/repo1",
                    filters=EventDependencyFilter(
                        data=[
                            DataFilter(
                                path="headers.X-Github-Event",
                                type="string",
                                value=["push"]
                            )
                        ]
                    )
                )
            ],
            triggers=[
                TriggerTemplate(
                    name="some-workflow-trigger",
                    argoWorkflow=ArgoWorkflowTrigger(
                        operation="submit",
                        source=Workflow(
                            api_version="argoproj.io/v1",
                            kind="Workflow",
                            metadata={"generateName": "some-workflow-"},
                            spec={
                                "workflowTemplateRef": {
                                    "name": "some-workflow-template"
                                }
                            }
                        )
                    )
                )
            ],
        )
    )
```

**Reply by elliotgunton** [reply]
Hey @gentry-au, sorry I missed this message - we don't currently have a way in Hera to connect Argo Events and Workflows, as we only have the spec generated. If you'd like to add an issue we can prioritise it from there! 🚀

**Reply by henrique-lh** [reply]
Hello, @elliotgunton! Are there any updates on the possibility of using Argo Events in Hera?
